### PR TITLE
Implementation a average age function

### PR DESCRIPTION
**Pull request by AnnaLangner**
Implementation a average age function

### CODE REVIEW

**Review by klangner — APPROVED** [review]

**Comment by AnnaLangner** [comment]
I changed my age calculation method, which solution is better?

File: script/main.py [hunk 1]
```diff
@@ -212,6 +212,37 @@ def percentage(conn, gender):
     print('Percentage of women: ' , percentage_of_women, '%')
   else:
     print('You are entering the wrong gender')
+
+
+def average_age(conn, gender):
```
Comment (klangner) [inline script/main.py:217]: In general calculating the age is always better then getting age from the database. So this is better approach.

File: script/main.py [hunk 2]
```diff
@@ -212,6 +212,37 @@ def percentage(conn, gender):
...
+  gender_rows = select_all_gender_and_age(conn)    
+  male = 0
+  female = 0
+  sum_of_age_male = 0
+  sum_of_age_female = 0
+  
+  for item in gender_rows:
+    full_date_of_birth = datetime.strptime(item[1], '%Y-%m-%dT%H:%M:%S.%fZ')
+    date_of_birth = full_date_of_birth.date()
+    today = date.today()
```
Comment (klangner) [inline script/main.py]: today should be out of the loop. 
It is waste of computer resources to get time inside the loop

File: script/main.py [hunk 3]
```diff
@@ -212,6 +212,37 @@ def percentage(conn, gender):
...
+    age = int((today - date_of_birth).days/365)
```
Comment (klangner) [inline script/main.py]: There is better way to get number of years between 2 dates. Check:
https://stackoverflow.com/questions/4436957/pythonic-difference-between-two-dates-in-years

File: script/main.py [hunk 4]
```diff
@@ -212,6 +212,37 @@ def percentage(conn, gender):
...
+    today = date.today()
+    age = int((today - date_of_birth).days/365)    
+    if item[0] == 'male':
+      male = male + 1
+      sum_of_age_male = sum_of_age_male + age
+    else:
+      female = female + 1
+      sum_of_age_female = sum_of_age_female + age
+
+  average_age_of_men = int(sum_of_age_male/male)
```
Comment (klangner) [inline script/main.py]: why you convert it to int? It is better to have it as float.

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py [hunk 5]
```diff
@@ -212,6 +212,37 @@ def percentage(conn, gender):
...
+    yaer_of_birth = full_date_of_birth.date().year  
+    age = today - yaer_of_birth
+    if item[0] == 'male':
+      male = male + 1
+      sum_of_age_male = sum_of_age_male + age
+    else:
+      female = female + 1
+      sum_of_age_female = sum_of_age_female + age
+
+  average_age_of_men = int(sum_of_age_male/male)
```
Comment (klangner) [inline script/main.py]: You shouldn't convert it to int here. float is better.

File: script/main.py [hunk 6]
```diff
@@ -212,6 +212,37 @@ def percentage(conn, gender):
...
+  average_age_of_women = int(sum_of_age_female/female)
+  average_age_overall = int((sum_of_age_male + sum_of_age_female)/(male + female))
+
+  if gender == 'male':
+    print('Average age of men: ', average_age_of_men, ' years')
```
Comment (klangner) [inline script/main.py]: if you leave it as float, then here you will get lots of digits after the dot. Learn how to print it correctly in python with for example with precision of 2 digits after the dot.

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py [hunk 7]
```diff
@@ -212,6 +212,37 @@ def percentage(conn, gender):
...
+    else:
+      female = female + 1
+      sum_of_age_female = sum_of_age_female + age
+
+  average_age_of_men = float(sum_of_age_male/male)
+  average_age_of_women = float(sum_of_age_female/female)
+  average_age_overall = float((sum_of_age_male + sum_of_age_female)/(male + female))
+
+  if gender == 'male':
+    print('Average age of men: ', round(average_age_of_men, 2), ' years')
```
Comment (klangner) [inline script/main.py]: better check string.format
https://pyformat.info/

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py (hunk 7 above)
Comment (AnnaLangner) [reply script/main.py]: Something like that?

**Review by klangner — APPROVED** [review]

**Pull request merged**